src/pymovements/dataset/dataset_definition.py

```diff
@@ -254,3 +321,41 @@ def to_yaml(self, path: str | Path, exclude_private: bool = True) -> None:
 
         with open(path, 'w', encoding='utf-8') as f:
             yaml.dump(data, f, sort_keys=False)
+
+    @property
+    def has_resources(self) -> _HasResourcesIndexer:
+        """Checks for resources in :py:attr:`~pymovements.dataset.DatasetDefinition.resources`.
+
+        This read-only property checks if there are any resources set in
+        :py:attr:`~pymovements.dataset.DatasetDefinition.resources`. It can be used as a `bool` or
+        as an indexable class. In a boolean context it checks if there are any resources set in the
+        :py:cls:`~pymovements.dataset.DatasetDefinition`. Furthermore, you can index the property
+        to check if there are any resources set for a given content type.
+
+        Examples
+        --------
+        This custom :py:cls:`~pymovements.dataset.DatasetDefinition` has no resources defined:
+        >>> import pymovements as pm
+        >>> my_definition = pm.DatasetDefinition('MyDatasetWithoutOnlineResources', resources=None)
+        >>> my_definition.has_resources
+        False
+
+        A :py:cls:`~pymovements.dataset.DatasetDefinition` from our
+        :py:cls:`~pymovements.dataset.DatasetLibrary` will usually have some online resources
+        defined:
+        >>> definition = pm.DatasetLibrary.get('ToyDataset')
+        >>> definition.has_resources
+        True
+
+        You can also check if a specific content type is contained in the resources:
+        >>> definition.has_resources['gaze']
+        True
+
+        In this definition there are gaze resources defined, but no precomputed events.
+        >>> definition.has_resources['precomputed_events']
+        False
+        """
+        # Resources may have changed, so update indexer before returning.
+        # A better way to update the resources would be through a resources setter property.
+        self._has_resources.set_resources(self.resources)
```

Comment:
Unfortunately we have do set the resources on each call, as we cannot guarantee that `self.resources` has not changed.

This could be improved by having a `DatasetResources` class with a `has_resources` property and then forwarding the call to `DatasetDefinition.resources.has_resources` in the property here instead.
related to #1120.